### ISSUE TITLE
[Bug]: Regression of `useResolvedPath` in 6.19.0

### ISSUE THREAD

**Issue by sgoll** [remix-run/react-router#11048]
### What version of React Router are you using?

6.19.0

### Steps to Reproduce

Use `useResolvedPath()` with relative path inside double-nested `<Route/>`.

See https://github.com/sgoll/react-router-test for a full reproduction with more details.

### Expected Behavior

`useResolvedPath()` should return the same path as in 6.18.0 and be independent of nested routes.

### Actual Behavior

When nested in routes within routes (double-nesting), `useResolvedPath()` returns one level more than expected; this is different from before (6.18.0) and different from the top-level behaviour.

The behaviour is also not consistent when adjusting to explicit relative paths like `./dolor` or `../dolor`. The first doesn't seem to change anything, and the second strips away two levels simultaneously.

### Screenshots

| This seems okay (6.18.0) | This seems wrong (6.19.0) |
|---|---|
| <img width="736" alt="react-router-6 18 0" src="https://github.com/remix-run/react-router/assets/1277035/83a3dba8-c1db-4e8e-8871-ecf68bb18851"> | <img width="736" alt="react-router-6 19 0" src="https://github.com/remix-run/react-router/assets/1277035/280b4c05-e3bb-4ed9-ba59-62d6a57b19e1"> |
| `useResolvedPath("dolor") -> {"pathname":"/ipsum/dolor",…}` | `useResolvedPath("dolor") -> {"pathname":"/ipsum/dolor/dolor",…}` |

**Comment by brophdawg11** [comment]
This is the intended behavior - what you're showing here was the previous buggy behavior that in splat routes, `useResolvedPath` was relative to the current route location without _without the splat_.  There's a bit of a longer explanation [here](https://github.com/remix-run/remix/issues/8054#issuecomment-1819380737) but the general idea in your reproduction is that when you do `useResolvedPath("dolor")` from the element rendered at `ipsum/*` while the URL is `/ipsum/dolor`, it will be relative to `/ipsum/dolor` and thus will append and become `/ipsum/dolor/dolor`

**Comment by chinfeng** [comment]
I have the same problem. Now I am unable to deal with navigate inside '/path/to/*' route ...

eg:

- previous: 
```javascript
const to = useResolvedPath('path/to');
const handle = () => navigate(to); // it will navigate to /base/path/to.
```

- now
```javascript
const to = useResolvedPath('path/to');
const handle = () => navigate(to); // it will navigate /base/path/to/path/to/path/to .....
```

**Comment by brophdawg11** [comment]
> Your link suggests that I could use useResolvedPath("../dolor") but that resolves to /dolor instead of /ipsum/dolor

This is based on how you've set up the route hierarchy.  In the linked example, the splat is a level all by itself, which makes it easier to drop just the splat portion by going up one level.  You have a static portion and a splat potion in your setup which means `..` drops both of them.  

So you could change from `<Route path="lorem/*" />` to `<Route path="lorem"><Route path="*" /></Route>` and then you can use `..` accordingly.

Otherwise I would leverage `useLocation`/`useParams` to replace the splat portion of the path with a new splat value: https://codesandbox.io/p/sandbox/gifted-sound-2683p2

> this symmetry seems broken

Try this on 6.20.0 - we found a few additional codepaths that needed the same update we made for `useResolvedPath` in #10983 and released those in 6.20.0 (via #11045): 
https://github.com/remix-run/react-router/blob/main/CHANGELOG.md#v6200
